crates/typst-library/src/foundations/str.rs

```diff
@@ -575,6 +575,10 @@ impl Str {
 
     /// Splits a string at matches of a specified pattern and returns an array
     /// of the resulting parts.
+    ///
+    /// When the empty string is used as a separator, it separates every
+    /// character in the string, along with the beginning and end of the
+    /// string.
```

Comment:
I'm not sure the second sentence makes it 100% clear that two empty strings result at the start and end.